[WFLY-19438] Add hook to disable Snappy compression

kabir opened the pull request:
https://issues.redhat.com/browse/WFLY-19438

kabir committed:
[WFLY-19438] Add hook to disable Snappy compression

kabir committed:
[WFLY-19438] Test to ensure that deployments fail if snappy is enabled and we don't support it

bstansberry commented on microprofile/reactive-messaging-smallrye/kafka/src/main/java/org/wildfly/microprofile/reactive/messaging/config/kafka/ssl/context/KafkaReactiveMessagingSslConfigProcessor.java:
```diff
@@ -6,11 +6,16 @@
+import java.util.Locale;
 import java.util.Map;
 
+import static org.wildfly.microprofile.reactive.messaging.config.kafka.ssl.context._private.MicroProfileReactiveMessagingKafkaLogger.LOGGER;
```
Order of import sections is reversed from what it should be.

bstansberry commented on microprofile/reactive-messaging-smallrye/kafka/src/main/java/org/wildfly/microprofile/reactive/messaging/config/kafka/ssl/context/_private/MicroProfileReactiveMessagingKafkaLogger.java:
```diff
@@ -33,4 +34,7 @@ public interface MicroProfileReactiveMessagingKafkaLogger extends BasicLogger {
     @Message(id = 2, value = "Could not find an Elytron client-ssl-context called: %s")
     IllegalStateException noElytronClientSSLContext(String ctx);
 
+    @Message(id = 3, value = "Snappy compression is not supported when running on Windows. The MicroProfile Config " +
```
s/Windows/Windows or OSX/g

bstansberry reviewed (changes requested):
Minor point about the error message.

It looks like your IDE import settings need a tweak. ;)

kabir committed:
[WFLY-19438] Code review fixes

The reorganisation of the imports touches more than what I changed, but might as well get it right

kabir commented:
@bstansberry Updated
